src/components/legend/style.js

```diff
@@ -288,6 +291,14 @@ module.exports = function style(s, gd) {
     function styleWaterfalls(d) {
         var trace = d[0].trace;
 
+        if(d[0]._distinct && trace.type === 'waterfall') {
```

Comment:
Now that we test for `trace.type === 'waterfall'` multiple times, let's make a variable and use it in those two places.
```js
var isWaterfall = trace.type === 'waterfall';
```

Reply:
Done in 67b1fa69210d9b38006d4b41205fb85770e3c835